consul/agents.go

```diff
@@ -31,38 +33,48 @@ func (a *ConcurrentAgents) GetAnyAgent() (*consulapi.Client, error) {
 	a.lock.Lock()
 	defer a.lock.Unlock()
 
-	for _, agent := range a.agents {
-		return agent, nil
+	if len(a.agents) > 0 {
+		return a.agents[a.getRandomAgentHost()], nil
 	}
 	return nil, fmt.Errorf("No agent available")
 }
 
-func (a *ConcurrentAgents) GetAgent(agentAddress string) (*consulapi.Client, error) {
+func (a *ConcurrentAgents) getRandomAgentHost() string {
+	hosts := []string{}
+	for host, _ := range a.agents {
+		hosts = append(hosts, host)
+	}
+	rand.Seed(time.Now().UTC().UnixNano())
```

Comment:
Seed is set every time this function is called. Better set it once in other case we could always end up with same host